examples/observe/server/main.go

```diff
@@ -29,20 +29,21 @@ func sendResponse(cc mux.Client, token []byte, subded time.Time, obs int64) erro
 		Body:    bytes.NewReader([]byte(fmt.Sprintf("Been running for %v", time.Since(subded)))),
 	}
 	var opts message.Options
-	var buf []byte
-	opts, n, err := opts.SetContentFormat(buf, message.TextPlain)
+	var cfbuf []byte
+	opts, n, err := opts.SetContentFormat(cfbuf, message.TextPlain)
 	if err == message.ErrTooSmall {
-		buf = append(buf, make([]byte, n)...)
-		opts, _, err = opts.SetContentFormat(buf, message.TextPlain)
+		cfbuf = append(cfbuf, make([]byte, n)...)
+		opts, _, err = opts.SetContentFormat(cfbuf, message.TextPlain)
 	}
 	if err != nil {
 		return fmt.Errorf("cannot set content format to response: %w", err)
 	}
+	var obsbuf []byte
```

Comment:
Using two buffers works, though i'd say the idiomatic way is to update the buffer head to be past the bytes written by the last option:

```go
var opts message.Options
var buf []byte
opts, n, err := opts.SetContentFormat(buf, message.TextPlain)
if err == message.ErrTooSmall {
	buf = append(buf, make([]byte, n)...)
	opts, n, err = opts.SetContentFormat(buf, message.TextPlain)
}
if err != nil {
	return fmt.Errorf("cannot set content format to response: %w", err)
}

buf = buf[n:]

if obs >= 0 {
	opts, n, err = opts.SetObserve(buf, uint32(obs))
```

Reply:
Updated as suggested. Shall I squash?

Reply:
Yes, squash and you can merge, good job!